aiida_vasp/calcs/workchains/converge.py

```diff
@@ -0,0 +1,1369 @@
+        spec.input('verify.max_iterations', valid_type=get_data_class('int'), required=False)
+        spec.input('verify.clean_workdir', valid_type=get_data_class('bool'), required=False)
+        spec.input('relax.incar', valid_type=get_data_class('parameter'), required=False)
+        spec.input('relax.perform', valid_type=get_data_class('bool'), required=False, default=get_data_class('bool')(True))
+        spec.input('relax.positions', valid_type=get_data_class('bool'), required=False, default=get_data_class('bool')(True))
+        spec.input('relax.shape', valid_type=get_data_class('bool'), required=False, default=get_data_class('bool')(False))
+        spec.input('relax.volume', valid_type=get_data_class('bool'), required=False, default=get_data_class('bool')(False))
+        spec.input('relax.convergence.on', valid_type=get_data_class('bool'), required=False, default=get_data_class('bool')(False))
+        spec.input('relax.convergence.absolute', valid_type=get_data_class('bool'), required=False, default=get_data_class('bool')(False))
+        spec.input('relax.convergence.max_iterations', valid_type=get_data_class('int'), required=False, default=get_data_class('int')(5))
+        spec.input(
+            'relax.convergence.shape.lengths', valid_type=get_data_class('float'), required=False,
+            default=get_data_class('float')(0.1))  # in cartesian coordinates
+        spec.input(
+            'relax.convergence.shape.angles', valid_type=get_data_class('float'), required=False,
+            default=get_data_class('float')(0.1))  # in degree in the cartesian system
+        spec.input(
+            'relax.convergence.volume', valid_type=get_data_class('float'), required=False,
+            default=get_data_class('float')(0.01))  # in degree in the cartesian system
+        spec.input(
+            'relax.convergence.positions', valid_type=get_data_class('float'), required=False,
+            default=get_data_class('float')(0.01))  # in degree in the cartesian system
+
+        spec.outline(
+            cls.initialize,
+            if_(cls.run_conv_calcs)(
+                while_(cls.run_pw_conv_calcs)(
+                    cls.init_pw_conv_calc,
+                    cls.init_next_workchain,
+                    cls.run_next_workchain,
+                    cls.results_pw_conv_calc
+                ),
+                cls.analyze_pw_conv,
+                while_(cls.run_kpoints_conv_calcs)(
+                    cls.init_kpoints_conv_calc,
+                    cls.init_next_workchain,
+                    cls.run_next_workchain,
+                    cls.results_kpoints_conv_calc
+                ),
+                cls.init_disp_conv,
+                while_(cls.run_pw_conv_disp_calcs)(
+                    cls.init_pw_conv_calc,
+                    cls.init_next_workchain,
+                    cls.run_next_workchain,
+                    cls.results_pw_conv_calc
+                ),
+                if_(cls.run_pw_conv_disp_calcs)(
```

Comment:
Isn't `cls.run_pw_conf_disp_calcs == FALSE` here by definition after `while_(cls.run_pw_conv_disp_calcs)(..)`?

Reply:
Indeed...fixed. I think I was going to protect it if we did not want displacements or volume changes. Of course, this does not really work, so I added another.